src/main/java/org/opensearch/jobscheduler/model/ExtensionJobParameter.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.jobscheduler.model;
+
+import java.io.IOException;
+import java.time.Instant;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+import org.opensearch.jobscheduler.spi.schedule.CronSchedule;
+import org.opensearch.jobscheduler.spi.schedule.IntervalSchedule;
+import org.opensearch.jobscheduler.spi.schedule.Schedule;
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+
+/**
+ * A {@link Writeable} ScheduledJobParameter used to transport job parameters between OpenSearch and Extensions
+ *
+ */
+public class ExtensionJobParameter implements ScheduledJobParameter, Writeable {
+
+    enum ScheduleType {
+        CRON,
+        INTERVAL
+    }
+
+    private String extensionJobName;
+
+    private Schedule extensionJobSchedule;
+
+    private Instant lastUpdateTime;
+
+    private Instant enabledTime;
+
+    private boolean isEnabled;
+
+    private Long lockDurationSeconds;
```

Comment:
Why is this `Long` and not `long`?  Introduces the possibility of it being `null` (vs. default 0) and would require special handling in the `Writable` implementation (e.g., `writeOptionalLong`) if null is a desired/possible value.  (In which case `@Nullable` annotation would be helpful.)

Reply:
meant to use `long`, thanks for catching this, I'll go ahead and make this change

Reply:
The `ExtensionJobParameter` implements the `ScheduledJobParameter` interface. Within this interface, the `lockDurationSeconds` and `jitter` values are nullable, in the case that a scheduled job does not require a lock, or if a job execution delay isn't necessary. 

Thank you for pointing this out though, I'll go ahead and modify the writeTo method so that it writes an optional long and double as well